src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestState.cs

```diff
@@ -139,6 +140,16 @@ public void PinReceiveBuffer(byte[] buffer)
             }
         }
 
+        public void DisposeCtrReadFromResponseStream()
+        {
+            if (CtrReadFromResponseStream != null)
+            {
+                WinHttpTraceHelper.Trace("WinHttpRequestState.DisposeCtrReadFromResponseStream: disposing ctr");
+                CtrReadFromResponseStream.Dispose();
+                CtrReadFromResponseStream = default(CancellationTokenRegistration);
```

Comment:
Good question.  I can't do the assignment to null since it is a value type and I got a compiler error.

But now the question is what should `CtrReadFromResponseStream != null` be replaced with.  Maybe the Dispose is idempotent and could be done regardless. I will find out.